src/audio/module_adapter/module_adapter.c

```diff
@@ -1082,10 +1080,8 @@ static int module_adapter_copy_dp_queues(struct comp_dev *dev)
 				       source_get_data_available(data_src));
 
 		err = source_to_sink_copy(data_src, data_sink, true, to_copy);
-		if (err) {
-			comp_err(dev, "DP to LL copy error status: %d", err);
-			return err;
-		}
+		if (err)
+			comp_dbg(dev, "DP to LL copy error status: %d", err);
```

Comment:
better calculate min from 3 factors:
```
MIN(
   source_get_min_available(following_mod_data_source), 
   source_get_data_available(data_src), 
   sink_get_free_size(data_sink)
)
```

The problem is that it may happen that the data are not processed later and "data_sink" becomes full. I missed this point when introducing fix described in line 1063. 

Adrian, better calculate this minimum to copy as much data as you can, source_to_sink_copy will fail completely if there's not enough space in sink

Reply:
Sure, that will work too and allows to keep error logs